controllers/storagecluster/provider_server.go

```diff
@@ -189,6 +192,38 @@ func (o *ocsProviderServer) ensureDeploymentReplica(actual, desired *appsv1.Depl
 	return nil
 }
 
+// updateStorageProviderEndpointInStatus updates the status.storageProviderEndpoint which is required for consumer to establish a connection with provider
+func (r *StorageClusterReconciler) updateStorageProviderEndpointInStatus(instance *ocsv1.StorageCluster, nodePort int32) error {
```

Comment:
Can we have this as a util that returns the address instead of setting the address on the status?
This way you can call it above and update the status as part of the service reconcile

Reply:
done